transformer_engine/jax/mlp.py

```diff
@@ -23,369 +23,56 @@
 from .sharding import with_sharding_constraint_by_logical_axes
 
 
-def gelu(x: jnp.ndarray):
-    """
-    Gelu
-    """
-    output = _gelu(x)
-    return output
-
-
-@partial(jax.custom_vjp)
-def _gelu(x: jnp.ndarray):
-
-    geglu_output, _ = _gelu_fwd_rule(x)
-
-    return geglu_output
-
-
-def _gelu_fwd_rule(x):
-    geglu_output = te_gelu(x)
-    return geglu_output, (x,)
-
-
-def _gelu_bwd_rule(ctx, g):
-    x, = ctx
-    assert x.dtype == g.dtype
-
-    dx = dgelu(g, x)
-    dx = jnp.reshape(dx, x.shape)
-    return (dx,)
+activation_lu_dict = {
+    ('gelu',): { 'fwd': gelu,
+                "bwd": dgelu },
+    ('gelu', 'linear'): { 'fwd': gated_gelu,
+                         'bwd': dgated_gelu }
+}
 
+activation_lu_fp8_dict = {
+    ('gelu',): { 'fwd': gelu_fp8,
+                'bwd': dgelu_dbias_cast_transpose },
+    ('gelu', 'linear'): { 'fwd': gated_gelu_fp8,
+                          'bwd': dgated_gelu_cast_transpose }
+}
 
-_gelu.defvjp(_gelu_fwd_rule, _gelu_bwd_rule)
 
-
-def geglu(x: jnp.ndarray):
+def activation_lu(x: jnp.ndarray, activation_type: Sequence[Union[str, Callable]]):
     """
-    Gated gelu
+    Activation Unit
     """
-    assert x.shape[-2] == 2    # Linear + GeLU
-
-    output = _geglu(x)
-
+    if 'linear' in activation_type:
+        assert x.shape[-2] == 2 # Linear + GeLU
```

Comment:
How about something like
```
if len(activation_type) > 1:
    assert x.shape[-2] == len(activation_type)
```
Not sure if there are more complicated gate/activation cases in the future